modules/install/k3s_master.sh

```diff
@@ -3,123 +3,129 @@
 ## Uncomment the following lines to enable debug mode
 #set -x
 # PS4='+(${LINENO}): '
+set -x
+echo "$@"
+
+node_os=${1}
+fqdn=${2}
+public_ip=${3}
+private_ip=${4}
+ipv6_ip=${5}
+install_mode=${6}
+version=${7}
+channel=${8}
+datastore_type=${9}
+datastore_endpoint=${10}
+server_flags=${11}
+rhel_username=${12}
+rhel_password=${13}
 
 create_directories() {
-  mkdir -p /etc/rancher/k3s
   sudo mkdir -p -m 700 /var/lib/rancher/k3s/server/logs
   mkdir -p /var/lib/rancher/k3s/server/manifests
 }
 
 create_config() {
-  local fake_fqdn="${1}"
-  local node_external_ip="${2}"
-
+  hostname=$(hostname -f)
+  mkdir -p /etc/rancher/k3s
   cat <<EOF >>/etc/rancher/k3s/config.yaml
 write-kubeconfig-mode: "0644"
 tls-san:
-  - ${fake_fqdn}
-node-external-ip: "${node_external_ip}"
+  - $fqdn
 cluster-init: true
+node-name: ${hostname}
 EOF
 }
 
 add_config() {
-  local server_flags="${1}"
+  if [ -n "$server_flags" ] && [[ "$server_flags" == *":"* ]]
+  then
+    echo "$server_flags"
```

Comment:
No need to echo server flags since we cat the config.yaml later on

Reply:
Updated